src/ASM/LR/ASMu2Dmx.C

```diff
@@ -993,28 +993,12 @@ bool ASMu2Dmx::refine (const LR::RefineData& prm, Vectors& sol)
   }
 
   if (doRefine(prm, refBasis.get())) {
-    for (const LR::Meshline* line : refBasis->getAllMeshlines())
-      for (size_t j = 0; j < m_basis.size(); ++j)
-        if (refBasis == m_basis[j])
-          continue;
-        else {
-          int mult = 1;
-          if (ASMmxBase::Type == ASMmxBase::REDUCED_CONT_RAISE_BASIS1 ||
-              ASMmxBase::Type == ASMmxBase::REDUCED_CONT_RAISE_BASIS2) {
-            if (line->multiplicity_ > 1)
-              mult = line->multiplicity_;
-            else
-              mult = (j == 0 && ASMmxBase::Type == REDUCED_CONT_RAISE_BASIS1) ||
-                     (j == 1 && ASMmxBase::Type == REDUCED_CONT_RAISE_BASIS2) ? 2 : 1;
-          }
-
-          if (line->span_u_line_)
-            m_basis[j]->insert_const_v_edge(line->const_par_,
-                                            line->start_, line->stop_, mult);
-          else
-            m_basis[j]->insert_const_u_edge(line->const_par_,
-                                            line->start_, line->stop_, mult);
-        }
+    for (size_t j = 0; j < m_basis.size(); ++j)
+      if (refBasis != m_basis[j]) {
+        int mult = (j == 0 && ASMmxBase::Type == REDUCED_CONT_RAISE_BASIS1) ||
+                   (j == 1 && ASMmxBase::Type == REDUCED_CONT_RAISE_BASIS2) ? 2 : 1;
+        this->copyRefinement(m_basis[j].get(), mult);
```

Comment:
Cosmetic, but maybe the following would be as readable?
```
  if ((j == 0 && ASMmxBase::Type == REDUCED_CONT_RAISE_BASIS1) ||
      (j == 1 && ASMmxBase::Type == REDUCED_CONT_RAISE_BASIS2))
    this->copyRefinement(m_basis[j].get(),2);
  else
    this->copyRefinement(m_basis[j].get(),1);

Reply:
done.